### PR TITLE
Installation and usage documentation

### PR DESCRIPTION
**Pull request by djaychela**
A first attempt at providing documentation by improving README.md - I've left the first section blank as I thought you could provide a better description of the project.

Hopefully the rest of it is useful - I've aimed it at someone who needs hand-holding on each step, hence the explicit instructions for each platform and hopefully helpful explanations.

Aside from the PR I did when I was with you, I've not really done anything like this, so apologies if I've not followed appropriate protocol here!

### CODE REVIEW

**Comment by djaychela** [comment]
Missed out the installation of musescore on linux, so will do a new PR with that included.

**Comment by djaychela** [comment]
Sorry, re-opened as I didn't realise that making changes to my master branch would update the PR... apologies!

File: README.md [hunk 1]
```diff
@@ -1,2 +1,61 @@
...
+
+where
+
+`filename` - the music_tutorial file to run (such as 1/scale1.py)
+
+`mode` is one of:
+
+"midi" to hear a MIDI version of the file
+"score" to see the score
+"text" will show the python objects that the file generates
```
Comment (geokala) [inline README.md]: It would be good to put a full example after this, e.g.
m210run 1/scale1.py midi

**Comment by geokala** [comment]
My other comments in a review-ey fashion don't really apply to specific lines, so:
1. Do you think this would be better done as an html page so that it's easier for people new to programming to follow it? (then it could have links to the actual tutorials, and maybe a bit less power-user level output)

2. Do you think there should be an advanced section telling users how to change the paths to the relevant executables (musescore/vlc/timidity/windows media player), or is that something that should be assumed as "If you're going that far off piste then you're expected to already be competent"?

**Comment by geokala** [comment]
Side note: Great work, and thanks- I'd probably have less comments and be pretty much good to go if I hadn't been thinking about the documentation side for the last couple of days!

File: README.md (hunk 1 above)
Comment (djaychela) [reply README.md]: Definitely - I meant to put something like that in there, but forgot - I think the example you give would be ideal as it should work straight away and be a good test.

**Comment by djaychela** [comment]
> My other comments in a review-ey fashion don't really apply to specific lines, so:
> 
>     1. Do you think this would be better done as an html page so that it's easier for people new to programming to follow it? (then it could have links to the actual tutorials, and maybe a bit less power-user level output)

I'm not sure - I tend to look at the README when it's on github and the markdown is rendered, so maybe if there's an automated way to turn README.md into an html file which would be included in the repo?

> 
>     2. Do you think there should be an advanced section telling users how to change the paths to the relevant executables (musescore/vlc/timidity/windows media player), or is that something that should be assumed as "If you're going that far off piste then you're expected to already be competent"?

I think the second comment is probably pretty relevant, but then maybe that's assuming that if you know _something_ then you know _everything_ about this - there may be some people whose executables are in the non-default location, and even just pointing them to the right parts of the code to experiment with may be enough hand-holding to get them to try it themselves?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,63 @@
...
+When using the MIDI mode, music_tutorial will generate a MIDI file in a temporary directory, and then launch the appropriate system MIDI player to allow the user to hear it, currently:
+
+Timidity on linux
+Windows Media Player on Windows
+VLC on macOS
+
+Timidity or VLC need to be installed prior to using them (see requirements section)
+
+### Score View (score)
+When using score mode, music_tutorial will generate a musicXML file in a temporary directory, and then launch musescore to allow the user to view, edit and play it.
```
Comment (geokala) [inline README.md:60]: Only remaining change I think would be to add a note that Musescore needs to be installed prior to using it (similarly to the MIDI section)- sorry, should've spotted that earlier. If you can make that change then I'll merge.

**Comment by geokala** [comment]
Re: HTML vs README: Yes, fair point on this being the first place that will be seen (especially via github). Besides, there isn't currently an HTML section so let's get the README to a good state and then consider adding HTML later, rather than leaving everything in a not-complete state. <smiles>

For pointing at the executable location, I think the error message might help with that, but I'll give it some thought as well, and maybe add something when I come to do the first section.

File: README.md (hunk 2 above)
Comment (djaychela) [reply README.md:60]: I've done that now (plus made reference to 'prerequisites' rather than 'requirements' to clarify the difference between the two).

**Review by geokala — APPROVED** [review]

**Comment by geokala** [comment]
Excellent, thanks for that- I'll merge now.

**Pull request merged**